build.rs

```diff
@@ -1,14 +1,109 @@
 extern crate cbindgen;
 
-#[cfg(feature = "cheader")]
 use std::env;
-#[cfg(feature = "cheader")]
+use std::ffi::OsStr;
+use std::ops::Deref as _;
 use std::path::Path;
+use std::process::Command;
+use std::process::Stdio;
+use std::str;
+
+use anyhow::bail;
+use anyhow::Context as _;
+use anyhow::Result;
+
+/// Format a command with the given list of arguments as a string.
+fn format_command<C, A, S>(command: C, args: A) -> String
+where
+    C: AsRef<OsStr>,
+    A: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    args.into_iter().fold(
+        command.as_ref().to_string_lossy().into_owned(),
+        |mut cmd, arg| {
+            cmd += " ";
+            cmd += arg.as_ref().to_string_lossy().deref();
+            cmd
+        },
+    )
+}
+
+/// Run a command with the provided arguments.
+fn run<C, A, S>(command: C, args: A) -> Result<()>
+where
+    C: AsRef<OsStr>,
+    A: IntoIterator<Item = S> + Clone,
+    S: AsRef<OsStr>,
+{
+    let mut instance = Command::new(command.as_ref())
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .args(args.clone())
+        .output()
+        .with_context(|| {
+            format!(
+                "failed to run `{}`",
+                format_command(command.as_ref(), args.clone())
+            )
+        })?;
+
+    if !instance.status.success() {
+        let code = if let Some(code) = instance.status.code() {
+            format!(" ({})", code)
+        } else {
+            String::new()
+        };
+
+        let _last = instance.stderr.pop();
+        let stderr = if let Ok(stderr) = str::from_utf8(&instance.stderr) {
+            format!(": {stderr}")
+        } else {
+            format!(": {:?}", &instance.stderr)
+        };
+
+        bail!(
+            "`{}` reported non-zero exit-status{code}{stderr}",
+            format_command(command, args),
+        );
+    }
+
+    Ok(())
+}
+
+/// Build `data/test.bin`
+fn build_dwarf_v4_test_bin(test_bin: &Path) {
+    let mut output = test_bin.to_path_buf();
+    assert!(output.set_extension("bin"));
+
+    // Ideally we'd use the `cc` crate here, but it seemingly can't be convinced
+    // to create binaries.
```

Comment:
Could you share the errors you got? I used the `cc` crate before. It should work in this case.

Reply:
The `cc` crate can't be used to create a binary judging from the API surface it provides. It can only compile source files into object files and link them to the main program or perhaps not. We could perhaps get away doing some tests on object files, but for others we probably will need a final binary, I suppose.